src/app/components/utils/normalizeInput.ts

```diff
@@ -0,0 +1,69 @@
+import type { NormalizedInput, RawInput } from '../../types/types';
+
+export default function handleInput(pupil: RawInput): NormalizedInput {
+  const preparedName = prepareNormalisation(pupil.name);
+  const firstName = preparedName[0];
+  const middleName =
+    preparedName.length < 2 ? '' : preparedName.slice(1, -1).join(' ');
```

Comment:
maybe for a more semantic code you could set the middle name to none if its not there